Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,150 @@
+                    self.checkIfSaved(comic: comic)
+                }
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingPreviousComicAlert = true
+                }
+                print("ComicsVM, fetching previous comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func fetchNextComic() {
+        fetchingService.fetchComic(fetchingURL: .specificComic(comic.num + 1)) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.checkIfSaved(comic: comic)
+                }
+            case .failure(let error):
+                DispatchQueue.main.async {
+                    self.isShowingNextComicAlert = true
+                }
+                print("ComicsVM, fetching next comic failed. Error: \(error)")
+            }
+        }
+    }
+    
+    func toggleDescription() {
+        isShowingExplanation.toggle()
+    }
+    
+    func toggleInfo() {
+        isShowingInfo.toggle()
+    }
+    
+    func searchComic(searchNum: String) {
+        if isShowingSearch {
+            // TODO: error warning if the input is not int
+            fetchingService.fetchComic(fetchingURL: .specificComic(Int(searchNum) ?? 1)) { result in
+                switch result {
+                case .success(let comic):
+                    DispatchQueue.main.async {
+                        self.comic = comic
+                        self.isShowingSearch = false
+                        self.checkIfSaved(comic: comic)
+                    }
+                case .failure(let error):
+                    print("ComicsVM, searching next comic failed. Error: \(error)")
+                }
+            }
+        } else {
+            isShowingSearch = true
+        }
+    }
+    
+    func saveAsFavourite(comic: Comic) {
+        if comics.contains(comic) {
+            return
+        } else {
+            comics.append(comic)
+            parsingService.encodingComics(comics: comics) { result in
+                switch result {
+                case .success(let encodedData):
+                    UserDefaults.standard.set(encodedData, forKey: UserFefaultsKey.userDefaultsKey)
+                    print("ComicVM, saving as favourite")
+                    self.isSaved = true
+                case .failure(let error):
+                    print("ComicVM, saving as favourite failed: \(error)")
+                }
+            }
+        }
+    }
+    
+    func getFavouriteComics() {
+        // get the JsonFile first and then transform it into a Comic
+        guard let jsonData = parsingService.getUserDefaultData() else {
+            return
+        }
+        parsingService.decodingComics(jsonData: jsonData) { result in
+            switch result {
+            case .success(let comics):
+                for comic in comics {
```

Comment:
Here, you iterate through the list of comics, but don't do anything with them. So it seems `checkIfSaved` will always set isSaved to false?